src/web/services/student.service.spec.ts

```diff
@@ -75,25 +73,151 @@ describe('StudentService', () => {
       courseid: 'CS3281',
       studentemail: 'johndoe@gmail.com',
     };
+    spyOn(spyHttpRequestService, 'post').and.stub();
 
     service.regenerateStudentCourseLinks(paramMap.courseid, paramMap.studentemail);
 
     expect(spyHttpRequestService.post)
         .toHaveBeenCalledWith(ResourceEndpoints.STUDENT_COURSE_LINKS_REGENERATION, paramMap);
   });
 
-  it('should execute GET when loading students in a course as CSV', () => {
-    const paramMap: Record<string, string> = {
-      courseid: 'CS3281',
-    };
+  it('should execute GET on course & students endpoints when generating student csv list', () => {
+    const courseId: string = 'CS3281';
+    const httpSpy: Spy = spyOn(spyHttpRequestService, 'get').and.returnValue(of(''));
+    const studentList: Observable<string> = service.loadStudentListAsCsv({ courseId });
+    expect(httpSpy).toBeCalledWith(ResourceEndpoints.COURSE, {
+      courseid: courseId,
+      entitytype: 'instructor',
+    });
+    studentList.subscribe(() => expect(httpSpy).toBeCalledWith(ResourceEndpoints.STUDENTS, {
+      courseid: courseId,
+    }));
+  });
 
-    const responseType: string = 'text';
+  it('should show course student list with section', () => {
+    const courseId: string = 'listWithSection';
+    const courseName: string = 'List With Section';
+    const students: Student[] =
+      [
+        {
+          courseId,
+          email: 'student1OfTypicalCourse@typicalCourse.com',
+          name: 'student1OfTypicalCourse',
+          joinState: JoinState.JOINED,
+          teamName: 'Team 1',
+          sectionName: 'Section A',
+        },
+        {
+          courseId,
+          email: 'student2OfTypicalCourse@typicalCourse.com',
+          name: 'student2OfTypicalCourse',
+          joinState: JoinState.JOINED,
+          teamName: 'Team 2',
+          sectionName: 'Section A',
+        },
+      ];
+    expect(service.processStudentsToCsv(courseId, courseName, students)).toMatchSnapshot();
+  });
 
-    service.loadStudentListAsCsv({
-      courseId: paramMap.courseid,
-    });
+  it('should show course student list without section', () => {
+    const courseId: string = 'listWithoutSection';
+    const courseName: string = 'List Without Section';
+    const students: Student[] =
+      [
+        {
+          courseId,
+          email: 'student1OfTypicalCourse@typicalCourse.com',
+          name: 'student1OfTypicalCourse',
+          joinState: JoinState.JOINED,
+          teamName: 'Team 1',
+          sectionName: 'None',
+        },
+        {
+          courseId,
+          email: 'student2OfTypicalCourse@typicalCourse.com',
+          name: 'student2OfTypicalCourse',
+          joinState: JoinState.JOINED,
+          teamName: 'Team 2',
+          sectionName: 'None',
+        },
+      ];
+    expect(service.processStudentsToCsv(courseId, courseName, students)).toMatchSnapshot();
+  });
+
+  it('should show course student list with special team name', () => {
```

Comment:
There's nothing special in this test that warrants it to be a different test case. Unlike existence/absence of section which will affect the section column, special (?) team name only affects the individual row. Just combine this with  one of the previous test cases. You can make clear in e.g. the full name or email what the test data is about.

Reply:
I think I'll rewrite the test cases to utilise the chaining http calls at the same time